### PR TITLE
Lazy loading + Flickering issue

### LINKED ISSUE #123: TPP-00123: When opening app, there should not be a flicker of screens while we check if user needs to onboard or not

### PR DESCRIPTION
**Pull request by kenneth-miura**
# Description

- Added lazy loading screens for any page that makes backend calls to load information from the backend
- Adding lazy loading screen to the initial screen prevents the welcome page flickering

image

Fixes #123 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (if required)
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: tpp-app/src/home/components/LoadingVisual.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import React from "react";
+import { ActivityIndicator, StyleSheet, Text, View } from "react-native";
+
+/**
+ *
```
Comment (yuemily24) [inline tpp-app/src/home/components/LoadingVisual.js]: One sec it's missing a wholistic function description LOL it's just a return description

File: tpp-app/src/home/pages/CycleHistoryScreen.js [hunk 2]
```diff
@@ -70,37 +71,54 @@ export default function CycleHistoryScreen({navigation}){
...
+                            intervals={currentIntervals[selectedYear]} 
+                            renderedYear={selectedYear}
+                            onPeriod={onPeriod}
+                        />
+                    </SafeAreaView>
+                </ImageBackground>
+            </SafeAreaView>
+        )
+    } else {
+        console.log("Im loading")
```
Comment (yuemily24) [inline tpp-app/src/home/pages/CycleHistoryScreen.js]:
```suggestion
```

**Review by yuemily24 — APPROVED** [review]
VERY Very tiny tiny tiny things LOOKS GOOD OTHERWISE

**Review by helen-li — APPROVED** [review]
nice sir

File: tpp-app/src/home/pages/CalendarScreen.js [hunk 3]
```diff
@@ -130,9 +136,16 @@ export default function CalendarScreen ({ route, navigation }) {
                             }
                         }
                     }
-                    setMarked(markedState => ({...markedState, ...newMarkedData}));
+                    promises.push(setMarked(markedState => ({...markedState, ...newMarkedData})));
                 }
             }
+
+            Promise.all(promises).then(
```
Comment (faye-tan) [inline tpp-app/src/home/pages/CalendarScreen.js:143]: yuh!

**Pull request merged**